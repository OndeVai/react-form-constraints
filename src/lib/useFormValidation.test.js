import React from 'react'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import {
  runMountedTest,
  runSubmit,
  assertValidSubmit,
  runShallowTest
} from './__test-utils__'
//import TestForm from './__test-utils__/TestForm'

configure({ adapter: new Adapter() })

describe('<TestForm/>', () => {
  it('renders without crashing', () => {
    runMountedTest()
  })

  it('matches snapshot', () => {
    runShallowTest(wrapper => {
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })

  describe('getFormProps()', () => {
    it('prevents consumer submit if invalid input', () => {
      runMountedTest(wrapper => {
        assertValidSubmit(wrapper, false)()
      })
    })

    it('calls consumer submit if valid input', () => {
      runMountedTest(wrapper => {
        wrapper.find('[name="inputVal"]').simulate('change', {
          target: { value: 'ttt@email.com', name: 'inputVal' }
        })
        wrapper.find('[name="selectVal"]').simulate('change', {
          target: { value: 'xxxx', name: 'selectVal' }
        })
        assertValidSubmit(wrapper, true)()
      })
    })

    it('calls e.preventDefault() on submit', () => {
      runMountedTest(wrapper => {
        const e = {
          preventDefault: jest.fn()
        }
        runSubmit(wrapper, e)()
        expect(e.preventDefault).toHaveBeenCalled()
      })
    })

    it('returns valid on submit', () => {
      runMountedTest(wrapper => {
        const e = {
          preventDefault: jest.fn()
        }
        runSubmit(wrapper, e)()
        expect(e.preventDefault).toHaveBeenCalled()
      })
    })


    //returns valid validation state based on form state

    //
  })

  describe('getFieldProps()', () => {
    it('adds the name prop that is passed in', () => {
      runShallowTest(wrapper => {
        const name = wrapper.find('[name="inputVal"]').props().name
        expect(name).toBe('inputVal')
      })
    })
  })
})
