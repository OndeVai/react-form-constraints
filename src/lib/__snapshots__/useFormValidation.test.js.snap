// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TestForm/> matches snapshot 1`] = `
<form
  id="test-form"
  noValidate={true}
  onSubmit={[Function]}
>
  <div>
    <label
      htmlFor="inputVal"
    >
      InputVal
    </label>
    <input
      name="inputVal"
      onBlur={[Function]}
      onChange={[Function]}
      required={true}
      type="email"
      value=""
    />
  </div>
  <div>
    <label
      htmlFor="selectVal"
    >
      selectVal
    </label>
    <input
      name="selectVal"
      onBlur={[Function]}
      onChange={[Function]}
      required={true}
      value=""
    />
  </div>
  <div>
    <button
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;
